app/helpers/basedOnRecommendationActivityManager.tsx

```diff
@@ -0,0 +1,55 @@
+import * as Cookies from 'js-cookie';
+import { Dispatch } from 'redux';
+import RecommendationAPI from '../api/recommendation';
+import { ActionCreators } from '../actions/actionTypes';
+import { getUserGroupName } from './abTestHelper';
+import { KNOWLEDGE_BASED_RECOMMEND_TEST } from '../constants/abTestGlobalValue';
+export const BASED_ACTIVITY_COUNT_COOKIE_KEY = 'basedActivityCount';
+
+const MAX_COUNT = 16;
+
+function setActionCount(count: number): number {
+  let nextCount;
+  if (count === MAX_COUNT) {
+    nextCount = 1;
+  } else {
+    nextCount = count + 1;
+  }
+
+  Cookies.set(BASED_ACTIVITY_COUNT_COOKIE_KEY, String(nextCount));
+  return nextCount;
+}
+
+export function addPaperToRecommendationPool(isLoggedIn: boolean, paperId: number, actionArea: string) {
```

Comment:
The helper method name and the api method name are the same and seem to be confusing.

It would be better to distinguish each other with a more specific name.

(Because it is often possible to make mistakes when importing functions.)